addons/base_setup/models/base_setup.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+from openerp import _, api, fields, models
+
+
+# Specify Your Terminology will move to 'partner' module
+class SpecifyPartnerTerminolgy(models.TransientModel):
+    _name = 'base.setup.terminology'
+    _inherit = 'res.config'
+
+    partner = fields.Selection([
+        ('Customer', 'Customer'),
+        ('Client', 'Client'),
+        ('Member', 'Member'),
+        ('Patient', 'Patient'),
+        ('Partner', 'Partner'),
+        ('Donor', 'Donor'),
+        ('Guest', 'Guest'),
+        ('Tenant', 'Tenant')], string='How do you call a Customer',
+        required=True,
+        default='Customer')
+
+    def make_translations(self, name, type, src, value, res_id=0):
+        IrTranslation = self.env['ir.translation']
+        UserLang = self.env.user.lang
+        existing_translations = IrTranslation.search([
+            ('name', '=', name),
+            ('lang', '=', UserLang),
+            ('type', '=', type),
+            ('src', '=', src),
+            ('res_id', '=', res_id)])
+        if existing_translations:
+            existing_translations.write({'value': value})
+        else:
+            IrTranslation.create({'name': name, 'lang': UserLang, 'type': type, 'src': src, 'value': value, 'res_id': res_id})
+        return {}
+
+    @api.multi
+    def execute(self):
+        def _case_insensitive_replace(ref_string, src, value):
+            import re
+            pattern = re.compile(src, re.IGNORECASE)
+            return pattern.sub(_(value), _(ref_string))
+        IrModelFields = self.env['ir.model.fields']
+        IrUiMenu = self.env['ir.ui.menu']
+        IrActionsActWindow = self.env['ir.actions.act_window']
+        # translate label of field
+        for field in IrModelFields.search([('field_description', 'ilike', 'Customer')]):
```

Comment:
why need to check terminology only for "Customer"? . what is purpose of partner selection field ?